h/views/api/decorators/client_errors.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+"""API view decorators for exception views"""
+
+from __future__ import unicode_literals
+
+from pyramid.httpexceptions import HTTPNotFound, HTTPUnsupportedMediaType
+
+from h.views.api.helpers.media_types import valid_media_types
+
+
+def unauthorized_to_not_found(wrapped):
+    """View decorator to convert all 403 exceptions to 404s"""
+
+    def wrapper(context, request):
+        # We convert all 403s to 404s—replace the current context with a 404
+        # FIXME: We should be more nuanced about when we do this
+        response = wrapped(_standard_not_found(), request)
+        return response
+
+    return wrapper
+
+
+def normalize_not_found(wrapped):
+    """View decorator to make 404 error messages more readable"""
+
+    def wrapper(context, request):
+        # Replace incoming 404 with one that has a sensible message
+        response = wrapped(_standard_not_found(), request)
+        return response
+
+    return wrapper
+
+
+def validate_media_types(wrapped):
+    """View decorator to make convert certain 4xx errors to 415s"""
+
+    def wrapper(context, request):
+        # If Accept has been set
+        if request.accept:
+            # At least one of the media types in Accept must be known to the app
+            ok_types = [
+                media_type
+                for media_type in request.accept
+                if media_type in valid_media_types()
+            ]
+            if not ok_types:
```

Comment:
I think these six lines can be rephrased more idiomatically like:

```python
if any([t in valid_media_types() for t in request.accept]):
```

Reply:
Thanks for this! This section got this way because I'd written it as a `filter` (in my local, default py2.7 env) and realize there was a py3 compatibility issue. I'll clean it up.